### PR TITLE
Separate ".(IsOne φ)→A" and "Partial φ A"

### LINKED ISSUE #6015: Pi types and Partial types should not be considered inter-convertible
`Partial φ A` is essentially defined to be `.(IsOne φ) → A`, and indeed Agda thinks these types are equal, as can be checked with the following snippet.

```agda
data Bool : Set where
  tt ff : Bool

data Idₛ {ℓ} (A : SSet ℓ) (a : A) : A → SSet ℓ where
  reflₛ : Idₛ A a a

module _ (i : I) where
  _ : Idₛ (SSet _) (Partial (i ∨ ~ i) Bool) (.(IsOne (i ∨ ~ i)) → Bool)
  _ = reflₛ
```

But `Partial φ A` isn't _literally_ the same as that pi-type, since the type `Partial` unfolds into has its domain marked `domFinite = True`, so that η-equality for elements of `Partial φ A` is checked by decomposing `φ` and checking that they agree on all possible substitutions that make `φ = i1`:

```agda
  _ : Idₛ (Partial (i ∨ ~ i) Bool) (primPOr i (~ i) (λ _ → tt) λ _ → tt) (λ _ → tt)
  _ = reflₛ
```

But this same tabulation is _not_ performed for functions-whose-domain-is-`IsOne`:

```agda
  _ : Idₛ (.(IsOne (i ∨ ~ i)) → Bool) (primPOr i (~ i) (λ _ → tt) (λ _ → tt)) (λ _ → tt)
  -- _ = reflₛ -- does not type-check
```

Conversion checking of terms is not invariant under definitional equality of the types they are being compared as. Should Agda really think that `(Partial (i ∨ ~ i) Bool)` and `(.(IsOne (i ∨ ~ i)) → Bool)` are the same type? The same thing, verbatim, goes for `PartialP` vs `(z : IsOne φ) → A z`.

I also think that finite-domain-Π shouldn't be printed in the same way as ordinary Πs — or better yet, prevent the normal form of `Partial` from being printed as a Π type, somehow.

### PR DESCRIPTION
**Pull request by andreasabel**
Fixes #6015. Clone of PR
- #6026

### CODE REVIEW

**Comment by plt-amy** [comment]
CI is failing because the Windows job gets cancelled after an hour: not sure what's up with that...

**Comment by andreasabel** [comment]
> CI is failing because the Windows job gets cancelled after an hour: not sure what's up with that...

Yes, this one is mysteriously broken:
- #6128 

If you rebase onto `master` it won't be run anymore.  But you can also ignore the failure.

File: src/full/Agda/Syntax/Translation/ConcreteToAbstract.hs [hunk 1]
```diff
@@ -1090,13 +1090,16 @@ instance ToAbstract C.TypedBinding where
...
     t' <- toAbstractCtx TopCtx t
     tac <- traverse toAbstract $
-             case List1.mapMaybe (bnameTactic . C.binderName . namedArg) xs of
-               []      -> Nothing
-               tac : _ -> Just tac
-               -- Invariant: all tactics are the same
-               -- (distributed in the parser, TODO: don't)
+      -- Invariant: all tactics are the same
+      -- (distributed in the parser, TODO: don't)
+      case List1.mapMaybe (bnameTactic . C.binderName . namedArg) xs of
```
Comment (andreasabel) [inline src/full/Agda/Syntax/Translation/ConcreteToAbstract.hs:1095]: Note that this case distinction just implements `listToMaybe` from `base`.

**Review by andreasabel — COMMENTED** [review]
I skimmed this PR, LGTM!
(GitHub does not allow me to "Approve" it because I am now the formal author of this PR.)

**Pull request merged**